rfcs/013-release_deployment_tracking/README.md

```diff
@@ -4,396 +4,460 @@
 
 ## Background
 
-We build container images as artifacts for deployment, they are labelled with the git ref at the code point the artifact was packaged.
+We run a lot of applications. People are regularly making changes and deploying new code. In order that we don't get confused about what code is deployed where we should track what is deployed where, by whom, and for what reason.
 
-Deploying container images to production at present moves all container images forward to the latest built version.
+The build/release/deployment process can be described as follows:
 
-## Problem statement
+![Simple overview](simple_flow.png)
 
-We wish to track multiple environments with container images at different versions, and have an audit trail for deployments to those environments.
+A high level view of infrastructure includes:
 
-A deployment mechanism to take a container image for a service within a project and run that in a given environment is deliberately not described here.
+- A service that creates build artifacts from a given version of the codebase, e.g. creating a Docker image (a *build environment*)
+- A store for the created artifacts, e.g. Docker images (an *artifact store*)
+- An environment where services can run, e.g. ECS or Kubernetes (a *deployment environment*)
+- A database that tracks what version of each application is running
 
-SSM parameters are heavily favoured as they provide a data source for terraform.
+![Infrastructure overvier](high_level_infra.png)
 
-## Proposed Solution
+### Glossary
 
-This solution describes a set of steps to track build, release and deployment but is deliberately agnostic of deployment mechanism. In addition this RFC **does not** describe a mechanism for reporting build or deployment status.
+- **project:** The top level, consisting of one or more **service set**'s. This might indicate whole product and should be a single git repository, e.g. the catalogue project.
 
-### Build artifacts
+- **service:** Performs a distinct function within a **project**. e.g. id_minter, requests API, Front-end content app.
```

Comment:
I completely understand what you mean here, but it may not be clear to somebody who's unfamiliar with our stuff already.

How about:

“This could be a single step of a multi-stage pipeline, an API application, or a front-end content app.”